crates/astria-grpc-mock/src/response.rs

```diff
@@ -119,6 +138,26 @@ impl Clone for MockResponse {
     }
 }
 
+pub struct ResponseTemplate {
+    response: Box<dyn Respond>,
+    delay: Option<Duration>,
+}
+
+impl ResponseTemplate {
+    pub(crate) fn respond(
+        &self,
+        req: &tonic::Request<AnyMessage>,
+    ) -> (ResponseResult, Option<Duration>) {
+        (self.response.respond(req), self.delay)
+    }
+
+    #[must_use]
+    pub fn set_delay(mut self, delay: Option<Duration>) -> Self {
```

Comment:
Nit: I think I'd change this to `fn set_delay(mut self, delay: Duration) -> Self {`

This makes the calls in conductor's mocks slightly less elegant, but IMO makes more sense from a user's point of view.